api/libproject/devstart.rb

```diff
@@ -88,6 +88,128 @@ def dev_up(*args)
   :fn => lambda { |*args| dev_up(*args) }
 })
 
+def setup_local_environment()
+  root_password = ENV["MYSQL_ROOT_PASSWORD"]
+  ENV.update(Workbench::read_vars_file("db/vars.env"))
+  ENV["DB_HOST"] = "127.0.0.1"
+  ENV["MYSQL_ROOT_PASSWORD"] = root_password
+  ENV["DB_CONNECTION_STRING"] = "jdbc:mysql://127.0.0.1/workbench?useSSL=false"
+  ENV["PUBLIC_DB_CONNECTION_STRING"] = "jdbc:mysql://127.0.0.1/public?useSSL=false"
+end
+
+def run_local_migrations()
+  setup_local_environment
+  # Runs migrations against the local database.
+  common = Common.new
+  Dir.chdir('db') do
+    common.run_inline %W{./run-migrations.sh main}
+    common.run_inline %W{./run-migrations.sh data local}
+  end
+  Dir.chdir('db-cdr') do
+    common.run_inline %W{./generate-cdr/init-new-cdr-db.sh --cdr-db-name cdr}
+    common.run_inline %W{./generate-cdr/init-new-cdr-db.sh --cdr-db-name public}
+  end
+  common.run_inline %W{gradle :tools:loadConfig -Pconfig_file=../config/config_local.json}
+end
+
+Common.register_command({
+  :invocation => "run-local-migrations",
+  :description => "Runs DB migrations with the local MySQL instance; does not use docker. You must set MYSQL_ROOT_PASSWORD before running this.",
+  :fn => lambda { |*args| run_local_migrations() }
+})
+
+def start_local_api()
+  setup_local_environment
+  common = Common.new
+  common.status "Starting API server..."
+  common.run_inline %W{gradle appengineStart}
+end
+
+Common.register_command({
+  :invocation => "start-local-api",
```

Comment:
Likewise, seems like it might be tricky at a glance to figure out which of these commands I'm meant to use (vs `dev-up` and `run-api`). Perhaps we could parameterize run-api somehow? Or rename this? Or split these apart to be clearer?

I guess one thing to keep in mind is that eventually we may want to run `run-local-api-tests` independently on our workstation.

Reply:
See above. I'll try to give this some thought.